tools/inc/stljobs.h

```diff
@@ -0,0 +1,606 @@
+    unsigned long dwFlagsAndAttributes, HANDLE hTemplateFile) {
+    handle<invalid_handle_value_policy> result{CreateFileW(lpFileName, dwDesiredAccess, dwShareMode,
+        lpSecurityAttributes, dwCreationDisposition, dwFlagsAndAttributes, hTemplateFile)};
+    if (!result) {
+        api_failure("CreateFileW");
+    }
+
+    return result;
+}
+
+inline handle<invalid_handle_value_policy> create_named_pipe(LPCWSTR lpName, DWORD dwOpenMode, DWORD dwPipeMode,
+    DWORD nMaxInstances, DWORD nOutBufferSize, DWORD nInBufferSize, DWORD nDefaultTimeOut,
+    LPSECURITY_ATTRIBUTES lpSecurityAttributes) {
+    handle<invalid_handle_value_policy> result{CreateNamedPipeW(lpName, dwOpenMode, dwPipeMode, nMaxInstances,
+        nOutBufferSize, nInBufferSize, nDefaultTimeOut, lpSecurityAttributes)};
+    if (!result) {
+        api_failure("CreateNamedPipeW");
+    }
+
+    return result;
+}
+
+
+const auto is_space = [](const wchar_t c) { return c == L' '; };
+
+[[nodiscard]] inline handle<null_handle_policy> create_kill_on_job_close_job() {
+    handle<null_handle_policy> hJob{CreateJobObjectW(nullptr, nullptr)};
+    if (!hJob) {
+        api_failure("CreateJobObjectW");
+    }
+
+    JOBOBJECT_EXTENDED_LIMIT_INFORMATION limits{};
+    limits.BasicLimitInformation.LimitFlags =
+        JOB_OBJECT_LIMIT_KILL_ON_JOB_CLOSE | JOB_OBJECT_LIMIT_DIE_ON_UNHANDLED_EXCEPTION;
+    if (!SetInformationJobObject(hJob.get(), JobObjectExtendedLimitInformation, &limits, sizeof(limits))) {
+        api_failure("SetInformationJobObject");
+    }
+
+    return hJob;
+}
+
+inline void put_self_in_job() {
+    auto hJob = create_kill_on_job_close_job();
+
+    // Put ourselves in that job
+    if (!AssignProcessToJobObject(hJob.get(), GetCurrentProcess())) {
+        api_failure("AssignProcessToJobObject");
+    }
+
+    // Purposely leak hJob to avoid terminating ourselves
+    (void) hJob.detach();
+}
+
+class no_input_pipe {
+public:
+    no_input_pipe() {
+        SECURITY_ATTRIBUTES inheritSa;
+        inheritSa.nLength              = sizeof(inheritSa);
+        inheritSa.lpSecurityDescriptor = nullptr;
+        inheritSa.bInheritHandle       = TRUE;
+        HANDLE read;
+        if (!CreatePipe(&read, hDevNull.out(), &inheritSa, 0)) {
+            api_failure("CreatePipe");
+        }
+
+        CloseHandle(read);
+    }
+
+    no_input_pipe(const no_input_pipe&) = delete;
+    no_input_pipe& operator=(const no_input_pipe&) = delete;
+
+    HANDLE get() const noexcept {
+        return hDevNull.get();
+    }
+
+    static const no_input_pipe& instance() {
+        static no_input_pipe instance_;
+        return instance_;
+    }
+
+private:
+    handle<null_handle_policy> hDevNull;
+};
+
+class tp_io {
+public:
+    tp_io() = default;
+
+    explicit tp_io(handle<invalid_handle_value_policy>&& fileHandle_, PTP_WIN32_IO_CALLBACK callback, void* const pv,
+        const PTP_CALLBACK_ENVIRON pcbe)

```

Comment:
`const PTP_CALLBACK_ENVIRON pcbe = nullptr`. Since we are going to use the default environment, making the parameter optional will remove the clutter at the point of use (and attempting to look up what is that nullptr last parameter do)